### PR TITLE
Added documentation about using asynchronous APIs in a C# native module

### PR DESCRIPTION
**Pull request by qmatteoq**
The current documentation about authoring native modules in C# covers only creating methods which leverage synchronous APIs. However, most of the APIs in the Universal Windows Platform are implemented using the asynchronous pattern. However, due to the way React Native for Windows works, you can't create such methods using the standard approach, but you must use a few helpers provided by the platform. I've updated the documentation to include these information.

### CODE REVIEW

**Comment by kikisaints** [comment]
@stmoy can you review? I believe Native Modules are still your area of expertise :)

**Comment by stmoy** [comment]
+ @jonthysell as well. Overall this looks good and reasonable to me!

**Comment by qmatteoq** [comment]
@kikisaints and @stmoy I probably have made a mistake 😐 I submitted a Pull Request to upgrade the TodosFeed sample to React Native 0.62 but, since this PR was still open, my work got added to this one instead creating a new PR. Sorry for this!

**Comment by harinikmsft** [comment]
> @kikisaints and @stmoy I probably have made a mistake 😐 I submitted a Pull Request to upgrade the TodosFeed sample to React Native 0.62 but, since this PR was still open, my work got added to this one instead creating a new PR. Sorry for this!

@qmatteoq  - what are the next steps on this PR?

**Comment by qmatteoq** [comment]
Hello @harinikmsft I have some work to do because the PR has been stale for too long and, since the master always needs to reflect the most recent bits, it has been exceeded by #208 since 0.63 has brought direct support for Task<T>.

I'm on vacation this week, but next week I will review the latest changes and include only the update for the TodosFeed demo.

**Comment by asklar** [comment]
@qmatteoq do you need anything/are you blocked on anything to move this forward?

**Comment by qmatteoq** [comment]
> @qmatteoq do you need anything/are you blocked on anything to move this forward?

Only by time :-) I'm swamped with Ignite this week, this item is in my "To-Do" list for next week. Thanks for your patience!

**Comment by asklar** [comment]
Looks like samples/rssreader/msbuild.binlog  snuck through, can you remove it? :)

File: docs/native-modules.md [hunk 1]
```diff
@@ -1,519 +1,519 @@
----
```
Comment (asklar) [inline docs/native-modules.md:1]: Are there any changes to this file or are there only whitespace changes? If no changes can you remove this file from this change so it doesn't cause merge conflicts with the other PRs? Thanks!

**Comment by qmatteoq** [comment]
I apologize @asklar, I did some changes to upgrade rssreader to 0.62 but I forgot that this PR was still open.
I'm going to close it and submit a new one, since this PR contains too many moving parts.

Thanks!

**Pull request closed without merging**